google-cloud-spanner/lib/google/cloud/spanner/results.rb

```diff
@@ -85,6 +86,7 @@ def rows
 
           fields = @metadata.row_type.fields
           values = []
+          cached_responses = []
```

Comment:
Is `cached_responses` bounded somewhere? My understanding is that the buffer should be bounded, and once full, data returned regardless of `resume_token`.